src/main/java/hudson/plugins/jacoco/JacocoBuildAction.java

```diff
@@ -359,4 +355,14 @@ public final PrintStream getLogger() {
 	    // does not run the construct and thus leaves the transient variables empty
 	    return System.out;
 	}
+
+	@Override
+	public void onAttached(Run<?, ?> run) {
+		this.owner = run;
```

Comment:
@jglick so I can remove the first parameter of this constructor: `public JacocoBuildAction(Run<?,?> owner, Map<CoverageElement.Type, Coverage> ratios, ...)` without worrying about any side effect. Other plugins currently are not using JaCoCo API.